common/src/main/java/org/jboss/hal/testsuite/page/config/MessagingPage.java

```diff
@@ -58,58 +58,63 @@ public void navigateToMessagingProvider() {
             navigation = new FinderNavigation(browser, StandaloneConfigEntryPoint.class)
                     .step(FinderNames.CONFIGURATION, FinderNames.SUBSYSTEMS);
         }
-        navigation.step(FinderNames.SUBSYSTEM, MESSAGING_SUBSYSTEM_LABEL);
-        navigation.step(SETTINGS_LABEL, MESSAGING_PROVIDER_LABEL);
+        return navigation.step(FinderNames.SUBSYSTEM, MESSAGING_SUBSYSTEM_LABEL);
     }
 
-    public void selectProvider(String provider) {
-        navigateToMessagingProvider();
-        navigation.step(MESSAGING_PROVIDER_LABEL, provider);
+    public void navigate() {
+        createNavigationToMessagingSubsystem().selectColumn();
     }
 
-    public void navigateToMessaging() {
-        selectProvider("default");
+    private void invokeActionOnMessagingProviderColumn(String action) {
+       FinderFragment finderFragment = createNavigationToMessagingSubsystem()
+                .step(SETTINGS_LABEL, MESSAGING_PROVIDER_LABEL)
+                .step(MESSAGING_PROVIDER_LABEL)
+                .selectColumn();
+       Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
+       finderFragment.invoke(action);
     }
 
-    public void selectView(String view) {
-        Row row = navigation.selectRow();
-        Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
-        row.invoke(view);
-        Application.waitUntilVisible();
+    private void invokeActionOnMessagingProviderRow(String action, String name) {
+       FinderFragment finderFragment = createNavigationToMessagingSubsystem()
+                .step(SETTINGS_LABEL, MESSAGING_PROVIDER_LABEL)
+                .step(MESSAGING_PROVIDER_LABEL, name)
+                .selectRow();
+       Console.withBrowser(browser).dismissReloadRequiredWindowIfPresent();
+       finderFragment.invoke(action);
     }
 
-    public void selectConnectionsView() {
-        selectView("Connections");
+    public void viewMessagingProvider(String name) {
```

Comment:
I'm missing `Application.waitUntilVisible();` in this method but anyway it seems to never be used so please remove this method.